### DISCUSSION TITLE
Listings mit mehr als 1000 Logs erzeugen Performanceprobleme

### DISCUSSION THREAD

**Discussion by Schapabande** [2Abendsegler/GClh#1938] [Ideas]
Hallo zusammen, ich liebe GC little helper.
Aber es hat Probleme bei Listings mit mehr als (sagen wir mal) 1000 Logs. Ich vermute, dies liegt daran, wenn man die Logs bzgl. V.I.P. untersuchen lässt. Es fällt einem z.B. bei den vielen Virtuals und Earthcaches von Dänemark auf.
Dann meldet mein Firefox immer wieder, ein Skript würde zu lange dauern und FF blockieren. Soll das Skript fortgesetzt oder gestoppt werden.
Meine Idee wäre, wenn dies möglich ist, die Zahl der Logs, die untersucht werden, konfigurativ einzuschränken. Z.B. also mittels 1000 festlegen, dass nur die letzten 1000 Logs für VIP und anderes untersucht wird.
Das ganze geht natürlich nur, wenn der Nachlademechanismus dies überhaupt zulässt.
Grüße von der Schapabande

**Comment by 2Abendsegler** [comment]
Hi,

ja, das ist blöd.

Eine solche Meldung ist mir auch schon mal untergekommen, allerdings nicht beim anfänglichen Laden des Listings. Ich habe mal etwas rumprobiert, kann das Problem aber nicht reproduzieren. Ich kann sogar das [derzeit dickste Listing](https://coord.info/GC189E5) mit über 41.000 Logs aufrufen. 

Ich glaube nicht, dass das Problem dadurch entsteht, dass die Logs untersucht werden. Ich vermute eher, dass es hier um das Laden der Logs geht. Aus irgendeinem Grund scheint das bei dir ziemlich langsam zu gehen. 

Ruf bitte mal dieses Listing [hier](https://coord.info/GC2CW4H) auf. Es enthält knapp 5000 Logs. Bitte schau dir mal die "VIP List" an und  stopp mal in etwa die Sekunden die es dauert, bis die "VIP List" angezeigt wird, also das "Loading ..." verschwunden ist.

Hast du eine Idee mit wieviel Megabit du Daten aus dem Internet lädst, also wie gut deine Leitung ist?
Hast du vielleicht eine genaue Bezeichnung deines Rechners und sein Alter parat? 

Schau mal ob du zu meinen Fragen ein paar Antworten hast. Danach sehen wir weiter.

LG Frank

**Comment by Schapabande** [comment]
Hallo Frank,also das Problem besteht definitiv auch bei dem Listing gc2cw4h.Schon nach kurzer Zeit meldet Firefox "Diese Seite verlangsamt Firefox. Halten Sie die Seite an, um den Browser tu beschleunigen."Das Laden der Seite geht nicht weiter, außer ich betätige "Anhalten" oder klicke auf das Kreuz um die Zeile wegzunekommen.Ok im Hintergrund macht er weiter, aber selbst nach 8 min ist das Listingbimmer noch nicht vollständig da.Ich verwende FF 93.0 (32 Bit)Windows 10 und an derxBandbreite liegt es definitiv nicht.Ich habe ca. 10 VIP angehakt, weiß gerade nicht, wo man dies prüfen kann.Ich habe das Problem definitiv nur bei langen Listings, sonst nie. Darum habe ich die Anzahl Logs im Verdacht.Notebook mit 16 GB Speicher, Intel Core i7-5500U2.40vGHzMan kann damit sehr gut entwickeln.Gerne genauer, wenn ich weiß was gebraucht wird.Provider KabelBW -> Unitymedia -> VodafoneGruß Peter --Diese Nachricht wurde von meinem Android Mobiltelefon mit GMX Mail gesendet.

**Comment by Schapabande** [comment]
Mit deaktiviertem Skript dauert es 5 sGruß Peter --Diese Nachricht wurde von meinem Android Mobiltelefon mit GMX Mail gesendet.

**Comment by 2Abendsegler** [comment]
Hallo Peter,

deine Ausstattung hört sich zuerst mal ok an. 

Wenn das Script deaktiviert ist, dann werden initial lediglich um die 30 Logs geladen. 
Bei mir ist es so, dass die 5000 Logs in weniger als 5 Sekunden geladen sind, so sollte es auch sein. Bei mehr als 8 Min ist irgend etwas total faul.

Kannst du mal bitte noch dieses Listing [hier](https://coord.info/GC23522) aufrufen. Es hat knapp 300 Logs. Schau mal bitte wieder in die "VIP List" rechts im Listing, wie lange es in etwa dauert bis dort das "Loading ..." verschwindet.

Und schau bitte mal was du unter [Show at least ... logs](https://www.geocaching.com/my/#GClhShowConfig#a#settings_show_all_logs) eingetragen hast.

**Comment by Schapabande** [comment]
Hallo Frank,super, dass du dich da reinhängst.1. Mein Rechner arbeitet gerade mit Akku  hat damit eine gewisse Leistungsreduzierung. Aber sls ITler kann ich das ganz gut einschätzen und irgendwas ist tatsächlich komisch.Also das Listing habe ich geladen. Beim ersten mal jann ich dir leine Zeit nennen, da dercVIP Beteich ja etwssd weiter unten eingebaut wird. Musstexslso erst mal hin scrollen und mittels F5 neu laden. Es sind zwischeb 7-9 s, bis das Loading verschwindet. Eingestellt ist die 0=all bei Show at least ... logs.Mein Profil von F ist schon sehr alt und ich habe es unter dem FF Portable laufen.Habe ein NoScript 11.2.11 laufen, die Seiten sind aber frei geschalten. Werde dies mal deaktivieten.Das Problem habe ich auch schon länger, habe es nur immer wieder ignoriert. Doch in Dänemark war es total lästig,  man gewöhnt sich sehr stark an manche Annehmlichkeiten.Ok, das Deaktivieren von NoScript bringt nix.Gruß Peter --Diese Nachricht wurde von meinem Android Mobiltelefon mit GMX Mail gesendet.

**Comment by 2Abendsegler** [comment]
Ok, da haben wir doch den Übertäter. "0" bedeutet, dass alle Logs angezeigt werden, wenn das tausend Logs sind, dann muss der Rechner schon ziemlich schuften. Stell mal bitte anstatt "0", "30" ein, dann sollte alles wie gewünscht funktionieren.

Es geht also nicht ums Laden der Logs, sondern um die Anzeige sehr vieler Logs.

---
Vielleicht sollten wir das so ändern, dass nur eine vernünftige Anzahl von Logs hier wählbar sind. Also vielleicht 100 oder 200 Logs. Mir waren die Auswirkungen nicht wirklich klar.
@capoaira was meinst du.

**Reply by capoaira** [reply]
Ich habe die Logs auch auf 0 eingestellt und kenne das Problem. Meist hat es mir geholfen auf "Anhalten" zu drücken. Aber es macht durchaus Sinn, das zu beschränken, denn alle Logs braucht man ja nie wirklich und wenn man runterscrollt werden ja automatisch weitere Logs geladen.

**Comment by Schapabande** [comment]
Ja, das scheint zu funktionieren. Das skript kann relativ schnell auch die VIP-Liste darstellen. Damit ist für mich erst einmal das Problem soweit behoben.
Was man dann aber nicht tun sollte, ist auf einen Logeintrag aus der VIP-Liste klicken. Also auf den Smiley, wo er dann an die entsprechende Stelle in den Logs springen will. Denn da kommt dann wieder dasselbe Szenario und er muss erst einmal die ganzen Logs bis dahin lesen ....
Grüße Peter von der Schapabande

**Reply by 2Abendsegler** [reply]
> Was man dann aber nicht tun sollte, ist auf einen Logeintrag aus der VIP-Liste klicken. Also auf den Smiley, wo er dann an die entsprechende Stelle in den Logs springen will.

Das stimmt. Das ist das Gleiche wie wenn man den Button "Show all logs" drückt. Die Darstellung so vieler Logs verursacht im Browser Probleme.
 
Aber schön, dass das eigentliche Problem behoben ist. Ich werde ein Issue aufmachen, um die Anzahl der beim Öffnen eines Listings anzuzeigenden Logs zu begrenzen.
